sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnSignatures.java

```diff
@@ -96,6 +98,9 @@ private static DoFnSignature parseSignature(Class<? extends DoFn<?, ?>> fnClass)
     }
     errors.checkNotNull(inputT, "Unable to determine input type");
 
+    Map<String, StateDeclaration> stateDeclarations = analyzeStateDeclarations(errors, fnClass);
```

Comment:
I was somewhat torn about whether to pull out the state parameters as part of the interface of `ProcessElementMethod`.